src/main/java/org/folio/note/NoteRepositoryImpl.java

```diff
@@ -0,0 +1,205 @@
+package org.folio.note;
+
+import static io.vertx.core.Future.failedFuture;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.NotFoundException;
+
+import org.folio.db.model.NoteView;
+import org.folio.rest.jaxrs.model.Note;
+import org.folio.rest.jaxrs.model.NoteCollection;
+import org.folio.rest.persist.PgExceptionUtil;
+import org.folio.rest.persist.PostgresClient;
+import org.folio.rest.persist.Criteria.Limit;
+import org.folio.rest.persist.Criteria.Offset;
+import org.folio.rest.persist.cql.CQLWrapper;
+import org.folio.rest.persist.interfaces.Results;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.z3950.zing.cql.cql2pgjson.CQL2PgJSON;
+import org.z3950.zing.cql.cql2pgjson.CQL2PgJSONException;
+import org.z3950.zing.cql.cql2pgjson.FieldException;
+
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import io.vertx.ext.sql.UpdateResult;
+
+@Component
+public class NoteRepositoryImpl implements NoteRepository {
+
+  private static final String NOTE_VIEW = "note_view";
+  private static final String NOTE_TABLE = "note_data";
+
+  private final Logger logger = LoggerFactory.getLogger(NoteServiceImpl.class);
+
+  @Autowired
+  private Vertx vertx;
+
+  @Override
+  public Future<NoteCollection> getNotes(String cqlQuery, int offset, int limit, String tenantId) {
+    logger.debug("Getting notes. new query:" + cqlQuery);
+    Future<NoteCollection> future = Future.succeededFuture(null);
+    return future.compose(o -> {
+      CQLWrapper cql;
+      try {
+        cql = getCQLForNoteView(cqlQuery, limit, offset);
+      } catch (CQL2PgJSONException e) {
+        throw new IllegalArgumentException("Failed to parse cql query");
+      }
+      return getNotes(tenantId, cql);
+    });
+  }
+
+  /**
+   * Saves a note record to the database
+   *
+   * @param note - current Note  {@link Note} object to save
+   */
+  @Override
+  public Future<Note> saveNote(Note note, String tenantId) {
+    initId(note);
+    Future<String> future = Future.future();
+    PostgresClient.getInstance(vertx, tenantId)
+      .save(NOTE_TABLE, note.getId(), note, future.completer());
+
+    return future.map(noteId -> {
+      note.setId(noteId);
+      return note;
+    });
+  }
+
+  /**
+   * Fetches a note record from the database
+   *
+   * @param id id of note to get
+   */
+  @Override
+  public Future<Note> getOneNote(String id, String tenantId) {
+    Future<Note> future = Future.future();
+    PostgresClient.getInstance(vertx, tenantId)
+      .getById(NOTE_VIEW, id, NoteView.class,
+        reply -> {
+          if (reply.succeeded()) {
```

Comment:
you don't need to process any exceptions apart from NotFound here. just pass them through